rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -135,6 +137,8 @@ class RecorderImpl
   rclcpp::Service<rosbag2_interfaces::srv::Snapshot>::SharedPtr srv_snapshot_;
   rclcpp::Service<rosbag2_interfaces::srv::SplitBagfile>::SharedPtr srv_split_bagfile_;
 
+  std::recursive_mutex state_transaction_mutex_;
```

Comment:
```suggestion
  std::mutex state_transition_mutex_;
```
IMO `state_transition_mutex` name better fit for its purpose